clang-tools-extra/test/clang-tidy/checkers/readability/use-builtin-literals-cpp17.cpp

```diff
@@ -0,0 +1,12 @@
+// RUN: %check_clang_tidy -std=c++17-or-later %s readability-use-builtin-literals %t
+
+void warn_and_fix() {
+
+(char)u8'a';
+// CHECK-MESSAGES: :[[@LINE-1]]:1: warning: use builtin literals instead of casts [readability-use-builtin-literals]
+// CHECK-FIXES: 'a';
+(wchar_t)u8'a';
+// CHECK-MESSAGES: :[[@LINE-1]]:1: warning: use builtin literals instead of casts [readability-use-builtin-literals]
+// CHECK-FIXES: L'a';
+
+}
```

Comment:
I understand why this is done like so, but it's unconventional to have the body of the function not indented. At first glance these looks like weird globals.

For the `CHECK-FIXES:`, I think doing `{{  }}L'a';` should deal with matching the whitespace indentation. There are certainly similar examples in other checks' test code.